tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -126,6 +127,320 @@ def type_signature(self):
                                   py_typecheck.type_string(type(self._value))))
 
 
+class IntrinsicStrategy(abc.ABC):
+  @classmethod
+  @abc.abstractmethod
+  def validate_executor_placements(cls, target_executors):
+    pass
+
+
+class DefaultIntrinsicStrategy(IntrinsicStrategy):
+
+  def __init__(self, parent_executor):
+    self.executor = parent_executor
+
+  @classmethod
+  def validate_executor_placements(cls, target_executors):
+    py_typecheck.check_type(target_executors, dict)
+    for k, v in target_executors.items():
+      if k is not None:
+        py_typecheck.check_type(k, placement_literals.PlacementLiteral)
+      py_typecheck.check_type(v, (list, executor_base.Executor))
+      if isinstance(v, list):
+        for e in v:
+          py_typecheck.check_type(e, executor_base.Executor)
+      for pl in [None, placement_literals.SERVER]:
+        if pl in target_executors:
+          ex = target_executors[pl]
+          if isinstance(ex, list):
+            pl_cardinality = len(ex)
+            if pl_cardinality != 1:
+              raise ValueError(
+                  'Unsupported cardinality for placement "{}": {}.'.format(
+                      pl, pl_cardinality))
+
+  @classmethod
+  def _check_arg_is_anonymous_tuple(cls, arg):
+    py_typecheck.check_type(arg.type_signature,
+                            computation_types.NamedTupleType)
+    py_typecheck.check_type(arg.internal_representation,
+                            anonymous_tuple.AnonymousTuple)
+
+  @tracing.trace
+  async def _place(self, arg, placement):
```

Comment:
could any of these helpers methods be moved to the executor? are they general enough to be reusable?

Reply:
they are only used in the intrinsic methods, so I'd say they should stay on IntrinsicStrategy